crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -946,6 +946,31 @@ pub mod common_conditions {
         move |mut removals: RemovedComponents<T>| removals.iter().count() != 0
     }
 
+    /// Generates a [`Condition`](super::Condition)-satisfying closure that
+    /// returns `true` if there are any entities with a component of the given
+    /// type changed. Newly added components are considered changed.
```

Comment:
This runs a potentially very expensive query (iterating through all instances of component T) inline in the scheduler. This blocks the scheduler from launching other systems and can be a very big performance footgun, particularly when there are a large number of entities in the world with T. At a minimum, this should be noted in the doc comments, and it should recommend directly using change detection in the system as the recommenced alternative.